packages/eas-cli/src/build/runBuildAndSubmit.ts

```diff
@@ -78,6 +79,12 @@ export async function runBuildAndSubmitAsync(projectDir: string, flags: BuildFla
   }[] = [];
   const buildCtxByPlatform: { [p in AppPlatform]?: BuildContext<Platform> } = {};
 
+  // Check only first buildprofile (there should be only one unique one)
+  // Warn only once
+  if (buildProfiles.length > 0) {
+    await checkBuildProfileConfigMatchesProjectConfigAsync(projectDir, buildProfiles[0]);
+  }
```

Comment:
- Apply the suggestion shared on Slack.
- I'd move it to `prepareAndStartBuildAsync`. There's an `exp` object available there so you wouldn't need to call `getConfig` in `checkBuildProfileConfigMatchesProjectConfigAsync`.